core/quantity/src/Parser.ts

```diff
@@ -796,6 +800,94 @@ export class Parser {
     return { ok: true, value: magnitude };
   }
 
+  private static parseRatioFormat(inString: string, spec: ParserSpec): QuantityParseResult {
+    if (!inString || !inString.includes(":"))
+      return { ok: false, error: ParseError.NoValueOrUnitFoundInString };
+
+    const parts = inString.split(":");
+    if (parts.length !== 2)
+      throw new QuantityError(QuantityStatus.UnableToConvertParseTokensToQuantity, `Invalid ratio: ${inString}, needs to be in form x:y`);
+
+    const numerator = parseFloat(parts[0]);
+    const denominator = parseFloat(parts[1]);
+
+    if (isNaN(numerator) || isNaN(denominator))
+      return { ok: false, error: ParseError.NoValueOrUnitFoundInString };
+    // TODO - Naron: there is also a QuantityStatus.NoValueOrUnitFoundInString
+
+    // special corner cases
+    if (spec.format.units && spec.outUnit){
+      if (spec.format.units[0][0].name === spec.outUnit.name && numerator === 0 && denominator === 1)
+        return { ok: true, value: 0.0 };
+      // if (spec.format.units[0][0].name is invert of spec.outUnit && numerator === 1 && denominator === 0)
+      //   return { ok: true, value: 0.0 };
+    } else {
+      throw new QuantityError(QuantityStatus.MissingRequiredProperty, "Missing presentation unit or persistence unit for ratio format.");
+    }
+
+    let magnitude = 0.0;
+    switch (spec.format.ratioType) {
+      case RatioType.OneToN:
```

Comment:
This can be more simple. I say we do not validate the RatioType here. We are parsing a ratio, so we should try and accept "2:1" as well as "4:2" or "8:4". It translates into the same persisted value. No need to enforce the RatioType here. Only use it in the Formatter.